src/server/game/Chat/Channels/Channel.cpp

```diff
@@ -704,6 +704,11 @@ void Channel::Say(ObjectGuid const& guid, std::string const& what, uint32 lang,
     if (sWorld->getBoolConfig(CONFIG_ALLOW_TWO_SIDE_INTERACTION_CHANNEL))
         lang = LANG_UNIVERSAL;
 
+    Player* player = ObjectAccessor::FindConnectedPlayer(guid);
+
+    if (!player->GetSession()->IsAddonRegistered(prefix))
```

Comment:
Hmm, you're probably right, although I added this condition and it's the only way the addon would get a response back from the broadcast channel when debugged.